src/data-structures/hash-table.js

```diff
@@ -41,25 +48,71 @@
 
     for (i = 0; i < str.length; i += 1) {
       character = str.charCodeAt(i);
+      /*jshint -W016 */
       hashCode = ((hashCode << 5) - hashCode) + character;
       hashCode = hashCode & hashCode;
+      /*jshint -W016 */
     }
 
     return hashCode;
   };
 
-  exports.HashTable.prototype.put = function (key, value) {
+  exports.HashTable.prototype.put = function (key, data) {
     var hashCode = this.hashCode(key);
-    this.elements[hashCode] = value;
+    var newNode = new Node(key, data);
+
+    if (this.elements[hashCode] === null) {
+      this.elements[hashCode] = newNode;
+    } else {
+      var first = this.elements[hashCode];
+
+      while (first.next !== null) {
+        first = first.next;
+      }
+
+      first.next = newNode;
+      newNode.prev = first;
+    }
   };
 
   exports.HashTable.prototype.get = function (key) {
     var hashCode = this.hashCode(key);
-    return this.elements[hashCode];
+
+    if (this.elements[hashCode] === null) {
+      return null;
+    } else if (this.elements[hashCode].next === null) {
+      return this.elements[hashCode];
+    } else {
+      var first = this.elements[hashCode];
+
+      while (first.key !== key) {
+        first = first.next;
+      }
+
+      return first;
+    }
   };
 
   exports.HashTable.prototype.remove = function (key) {
     var hashCode = this.hashCode(key);
-    this.elements.splice(hashCode, 1);
+
+    if (this.elements[hashCode] === null) {
+      return false;
+    } else if (this.elements[hashCode].next === null) {
+      this.elements.splice(hashCode, 1);
```

Comment:
Using `splice(hashCode, 1)` will remove the key from the array.
This means that if we have:
`[{ key: "foo", value: "baz" }, { key: "bar", value: "foobar" }]`, `hashCode` has returned `0` for key `foo` and `1` for key `bar`. If you want to remove `foo`, according to the current implementation you'll use: `splice(0, 1)`, this will change the `elements` array to:
`[{ key: "bar", value: "foobar" }]`. Next time you want to access the element with key `bar` and you invoke `hashCode(bar)`, it will return `1`, but since your array looks like `[{ key: "bar", value: "foobar" }]`, you'll try to access the index `1`, which value will be `undefined`.